File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.java
import java.util.*;
import java.io.*;

class Solution {
    public int solution(int[] nums) {
        int limit = nums.length / 2;
        
        // 중복 제거
        Set<Integer> numSet = new HashSet<>();
        
        for (int n : nums) {
            numSet.add(n);
        }
        
        if (numSet.size() > limit) {
            return limit;
        }
        
        else {
            return numSet.size();
        }
        
    }
}

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.py
## N/2 마리까지 가져가는 거 허용

def solution(nums):
    answer = 0
    
    poke_dic = {}
    
    N = len(nums) // 2
    
    for i in nums :
        poke_dic[i] = 1
    
    if len(poke_dic) > N :
        answer = N
    else :
        answer = len(poke_dic)
    return answer